homeworks/3/demo/src/main/java/com/example/demo/posts.java

```diff
@@ -0,0 +1,77 @@
+package com.example.demo;
+
+import com.google.gson.Gson;
+import jakarta.servlet.*;
+import jakarta.servlet.http.*;
+import jakarta.servlet.annotation.*;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+
+@WebServlet(name = "posts", value = "/posts")
+public class posts extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        HttpSession session = request.getSession(false);
+        if (session == null) {
+            response.setStatus(401);
+            return;
+        }
+
+        StringBuffer jb = new StringBuffer();
+
+        String line = null;
+        try {
+            BufferedReader reader = request.getReader();
+            while ((line = reader.readLine()) != null)
+                jb.append(line);
+        } catch (Exception e) {
+            System.err.println(e.getMessage());
+        }
+
+        Gson gson = new Gson();
+        Post post = gson.fromJson(jb.toString(),Post.class);
+
+
+
+
+

```

Comment:
There is no need for such big gaps of empty lines. They make your code longer and harder to read.